### PR TITLE
Refactoring duplicate/complex code

### PR DESCRIPTION
**Pull request by loren**
- Similar filter_from_fields in BusinessServiceProviderQuery and ItaOfficeLocationQuery
- Similar multi_match query with unneeded bool/must as there's only ever one child
- Created GenericScreeningList class to consolidate common behavior
- Moved similar Envirotech behavior into base_data superclass
- Moved similar TradeEvent behavior into superclass

### CODE REVIEW

**Comment by loren** [comment]
Calum, I refactored a bunch of classes resulting in about 280 fewer lines and a lot of Code Climate issues resolved (https://codeclimate.com/github/GovWizely/webservices/compare/code_climate_jan_2016#ratings). Tests still pass. Does this look reasonable?

**Comment by calumhalcrow** [comment]
Sounds great :smiley: I will review this during my work day today.

File: app/importers/screening_list/generic_screening_list.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module ScreeningList
+  class GenericScreeningList
```
Comment (calumhalcrow) [inline app/importers/screening_list/generic_screening_list.rb]: Maybe just `Base`?

File: app/importers/trade_event/trade_event_data.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module TradeEvent
+  class TradeEventData
```
Comment (calumhalcrow) [inline app/importers/trade_event/trade_event_data.rb]: `BaseData`?

I often wanted to rename the importers from *Data to *Importer. Didn't ever get around to it though.

**Comment by calumhalcrow** [comment]
Looks good, thanks for doing this :smiley: 

Let me know what you think of my naming suggestions. Otherwise, I'm happy to merge this.

**Comment by loren** [comment]
Thank you for looking it over! I made both name changes.

**Pull request merged**